src/app.rs

```diff
@@ -176,76 +152,86 @@ impl<Ms, Mdl, INodes: IntoNodes<Ms> + 'static, GMs: 'static> App<Ms, Mdl, INodes
-        // @TODO: Remove as soon as Webkit is fixed and older browsers are no longer in use.
-        // https://github.com/David-OConnor/seed/issues/241
-        // https://bugs.webkit.org/show_bug.cgi?id=202881
-        let _ = util::document().query_selector("html");
+        app.data
+            // @TODO_B we are trying to get rid of this field
+            .main_el_vdom
+            // TODO_B with the removal of MountType, this becomes argument-free. This needs looking at
+            .replace(Some(app.bootstrap_vdom()));
 
-        // Allows panic messages to output to the browser console.error.
-        console_error_panic_hook::set_once();
+        let mut orders = OrdersContainer::new(app.clone());
+
+        let new_model = init(
+            Url::current().skip_base_path(&Rc::clone(&app.cfg.base_path)),
+            &mut orders,
+        );
+        app.data.model.replace(Some(new_model));
+
+        routing::setup_popstate_listener(
+            enc!((app => s) move |closure| {
+                s.data.popstate_closure.replace(Some(closure));
+            }),
+            enc!((app => s) move |notification| s.notify_with_notification(notification)),
+            Rc::clone(&app.cfg.base_path),
+        );
+        routing::setup_hashchange_listener(
+            enc!((app => s) move |closure| {
+                s.data.hashchange_closure.replace(Some(closure));
+            }),
+            enc!((app => s) move |notification| s.notify_with_notification(notification)),
+            Rc::clone(&app.cfg.base_path),
+        );
+        routing::setup_link_listener(
+            enc!((app => s) move |notification| s.notify_with_notification(notification)),
+        );
 
-        AppBuilder::new(update, view)
+        orders.subscribe(enc!((app => s) move |url_requested| {
+            routing::url_request_handler(
+                url_requested,
+                Rc::clone(&s.cfg.base_path),
+                move |notification| s.notify_with_notification(notification),
+            )
+        }));
+
+        app.process_effect_queue(orders.effects);
+        app.rerender_vdom();
+        app
     }
 
-    /// Invoke your `update` function with provided message.
+    /// This runs whenever the state is changed, ie the user-written update function is called.
+    /// It updates the state, and any DOM elements affected by this change.
+    /// todo this is where we need to compare against differences and only update nodes affected
+    /// by the state change.
+    ///
+    /// We re-create the whole virtual dom each time (Is there a way around this? Probably not without
+    /// knowing what vars the model holds ahead of time), but only edit the rendered, `web_sys` dom
+    /// for things that have been changed.
+    /// We re-render the virtual DOM on every change, but (attempt to) only change
+    /// the actual DOM, via `web_sys`, when we need.
+    /// The model stored in inner is the old model; `updated_model` is a newly-calculated one.
```

Comment:
I think I've rewritten it to something more useful for readers / users. See current master.